specs/core/0_beacon-chain.md

```diff
@@ -1578,34 +1578,37 @@ For each `exit` in `block.body.exits`:
 
 ## Per-epoch processing
 
-The steps below happen when `state.slot % EPOCH_LENGTH == 0`.
+The steps below happen when `state.slot % EPOCH_LENGTH == EPOCH_LENGTH - 1`.
 
 ### Helpers
 
+* Let `current_epoch_start_slot = state.slot - EPOCH_LENGTH + 1`.
+* Let `previous_epoch_start_slot = state.slot - 2 * EPOCH_LENGTH + 1`.
```

Comment:
It seems now `state.slot=63` is the first valid slot number to perform per-epoch transtion? When `state.slot=63`, `previous_epoch_start_slot = state.slot - 2 * EPOCH_LENGTH + 1 = 63 - 2 * 64 + 1 = -64`.
1. I think we somehow assume that "slot number is an unsigned 64-bit integer"? In this special case, `-64` may not conform "the previous epoch start slot" context.
2. Might related to: https://github.com/ethereum/eth2.0-specs/issues/192

Reply:
I think putting a min on this value of 0 actually makes the logic work.